File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    
    for i in range(len(photo)):
        sum = 0
        for j in range(len(photo[i])):
            if photo[i][j] in name:
                # print(yearning[name.index(photo[i][j])])
                sum += yearning[name.index(photo[i][j])]
        answer.append(sum)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        for (int i=0; i<photo.length; i++){
            int count = 0;
            for (int j=0; j<photo[i].length; j++){
                for (int k = 0; k<name.length; k++){
                    if (!name[k].equals(photo[i][j])) continue;
                    count += yearning[k];
                    break;
                }
            }
            answer[i] = count;
        }
        return answer;
    }
}